src/services/pathContents/pull/dir/constants.ts

```diff
@@ -0,0 +1,148 @@
+import { z } from 'zod'
+
+import { MissingHeadReportSchema } from 'services/comparison'
+import { UnknownFlagsSchema } from 'services/impactedFiles/schemas'
+
+const MissingCoverageSchema = z.object({
+  __typename: z.literal('MissingCoverage'),
+  message: z.string(),
+})
+
+const UnknownPathSchema = z.object({
+  __typename: z.literal('UnknownPath'),
+  message: z.string(),
+})
+
+const RepositoryConfigSchema = z.object({
+  indicationRange: z
+    .object({
+      upperRange: z.number(),
+      lowerRange: z.number(),
+    })
+    .nullable(),
+})
+
+const PathContentFileSchema = z.object({
+  __typename: z.literal('PathContentFile'),
+  name: z.string(),
+  path: z.string().nullable(),
+  hits: z.number(),
+  misses: z.number(),
+  partials: z.number(),
+  lines: z.number(),
+  percentCovered: z.number(),
+  isCriticalFile: z.boolean(),
+})
+
+export type PathContentFile = z.infer<typeof PathContentFileSchema>
+
+const PathContentDirSchema = z.object({
```

Comment:
This is also fine since it is more explicit. 

You could do something like [this](https://github.com/codecov/gazebo/blob/01f688c7cbf6f27017f8f02ab44c944111c3c328/src/services/pathContents/branch/dir/useRepoBranchContents.tsx#L19) and combine the definitions if you wanted in the future. Up to you.